### ISSUE TITLE
Click events pass through to underlying DOM on selecting a date

### ISSUE THREAD

**Issue by RobbieLePommie** [dbushell/Pikaday#327]
When selecting a date, the click event is passed through to the underlying DOM: this has undesired effects happening under the calendar.

The event handler returns null (line 420) however this doesn't always stop the bubble.

Suggest adding e.preventDefault() before this, or allowing the function to continue to the bit where the event handling is killed (line 428 onwards)

**Comment by nathancahill** [comment]
@RobbieLePommie You can test this [branch](https://github.com/nathancahill/Pikaday/tree/capture-click-events) to see if it fixes your issue. There's a lot of different events that could bubble from the Pikaday div to body, so the `_captureEvent` listener would work with others (mousedown, mouseup, etc).

The `_onMouseDown` you mentioned at L420 is pretty specific, so it should probably be left as is. I'm not even sure this fix really fits in the scope of the lib so I probably won't submit a pull request. All other DOM elements on your page will bubble to body, so this seems like a pretty specific fix. What undesired effects are you having?

**Comment by RobbieLePommie** [comment]
Firstly the good news: that branch appears to fix the pass through problem and also the not-working-on-mobile issue. Thanks!

Minor validation errors:
  line 470  col 33  Missing semicolon.
  line 634  col 46  Missing semicolon.

***************

Slightly irrelevant, but here are/were the details:

There were two problems I noticed.

The first effect was that text under the page was selected (i.e. coloured blue) between where the field was the opened the picker and where the click occurred to select the date/close the picker if the click point was directly above selectable text. I noticed that more on desktop (Chrome, but also IE 11).

The other side effect was that the cursor/focus shifted to input fields where the click happened; so if the picker was over another input field, focus and cursor would end up inside that input field (assuming the date you clicked was exactly over the input field). 

(Unfortunately I'm not able to share the site as it's client related - but I used your branch and removed both my fixes and it appears good.)

Please merge and close. Thanks.

**Comment by RobbieLePommie** [comment]
Also please see https://github.com/dbushell/Pikaday/issues/301 regarding that branch: the "touchend" event is not cancelled.Minor thing, but should be added.

**Comment by nathancahill** [comment]
> the "touchend" event is not cancelled

Right, my point is that there are hundreds of events that could be passed down to the body. Just like any other DOM element. I think the preferable way to handle this would be to use the `container` option to render the calendar in another DOM element, which is setup beforehand to capture the specific events you're trying to prevent from bubbling to the body.

This fix isn't really applicable outside of that narrow scope, so I'm not submitting a pull request, but I'll leave my branch up for future reference.

**Comment by RobbieLePommie** [comment]
> Right, my point is that there are hundreds of events that could be passed down to the body.

Which is why I suggested (right at the top) of trapping the very event that you DON'T want passed down: the actual click / tap on the date itself. 

So you're basically saying that your fix here has unknown consequences? In which case I may as well resort back to the fix I'd implemented (described above) which prevented the actual date selection click bubbling up. Yes, it's specific (that you indicated is a bad thing) but it's targeting the one single known event that needs not to be passed on.

And this issue therefore remains for anyone else.

**Comment by goggledefogger** [comment]
I also ran into this, and would like to know how to fix. It's weird because the click handler works fine on my Windows 8.1 Chrome, but totally doesn't work on my friend's Windows 8.1 Chrome (both the same version of Chrome too). The _onClick and _onInputChange events fire for me, but not for him. How to fix?

**Comment by diegoleme** [comment]
This bug only happens in version 1.3.3

**Comment by hitchcockwill** [comment]
What's the latest on this issue? Is there a pull request or is this in the main branch?

**Comment by gamesover** [comment]
@hitchcockwill this issue does not happen to me now. I think this issue shall be closed.

**Comment by willyyang** [comment]
this is still happening..
